packages/data-broker/src/__tests__/services/dhis/DhisInputSchemeResolvingApiProxy.test.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { translateElementKeysInEventAnalytics } from '@tupaia/dhis-api';
+import { createApiProxyStub, createApiStub } from './DhisInputSchemeResolvingApiProxy.stubs';
+
+jest.mock('@tupaia/dhis-api');
+
+const api = createApiStub();
+const proxy = createApiProxyStub(api);
+
+describe('DhisInputSchemeResolvingApiProxy', () => {
+  describe('getAnalytics', () => {
+    it('should swap codes for dhisId if all specified', async () => {
+      await proxy.getAnalytics({
+        dataElementCodes: ['EL1', 'EL2'],
+        organisationUnitCodes: ['ORG1'],
+      });
+      return expect(api.getAnalytics).toHaveBeenCalledOnceWith({
+        dataElementIds: ['dhisId_el1', 'dhisId_el2'],
+        organisationUnitIds: ['dhisId_ou1'],
+        inputIdScheme: 'uid',
+      });
+    });
+
+    it('should not swap any codes for ids if some org units dont have dhisId', async () => {
+      await proxy.getAnalytics({
+        dataElementCodes: ['EL1', 'EL2'],
+        organisationUnitCodes: ['ORG1', 'ORG2'], // org2 not present in mapping table, so doesn't have dhis id
+      });
+      return expect(api.getAnalytics).toHaveBeenCalledOnceWith({
+        dataElementCodes: ['EL1', 'EL2'],
+        organisationUnitCodes: ['ORG1', 'ORG2'],
+      });
+    });
+
+    it('should not swap any codes for ids if some data elements dont have dhisId', async () => {
+      await proxy.getAnalytics({
+        dataElementCodes: ['EL1', 'EL3'], // EL3 does not have dhis id
+        organisationUnitCodes: ['ORG1'],
+      });
+      return expect(api.getAnalytics).toHaveBeenCalledOnceWith({
+        dataElementCodes: ['EL1', 'EL3'],
+        organisationUnitCodes: ['ORG1'],
+      });
+    });
+
+    it('should not swap any codes for ids if some data elements dont exist in data_source', async () => {
+      // TODO: remove after tupaia-backlog#663 is done
```

Comment:
@kael89 please keep an eye on merge order with your work on 663 - there are a few pieces to this WIP that will need to be cleaned up in this branch (if yours hits dev first) or your branch (if this hits dev first) 🙏